File: py/IEC61968/Assets/LabTestDataSet.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106
from IEC61968.Assets.AssetTestLab import AssetTestLab
from IEC61968.Assets.ProcedureDataSet import ProcedureDataSet
from IEC61968.Assets.Specimen import Specimen
from IEC61968.Assets.TestReason import TestReason

class LabTestDataSet(ProcedureDataSet):
    
    def __init__(self):
        super().__init__()
        self.conclusion = ""
        self.conclusion_confidence = ""
        self.reason_for_test = TestReason()
        self.test_equipment_id = ""
        self.specimen = Specimen()
        self.asset_test_lab = AssetTestLab()

File: java/IEC61968/Assets/LabTestDataSet.java
package IEC61968.Assets;

import IEC61970.Base.Domain.String;

/**
 * Results of testing done by a lab.
 * @author ppbr003
 * @version 1.0
 * @created 25-Dec-2023 8:45:22 PM
 */
public class LabTestDataSet extends ProcedureDataSet {

	/**
	 * Conclusion drawn from test results.
	 */
	public String conclusion;
	/**
	 * Description of confidence in conclusion.
	 */
	public String conclusionConfidence;
	/**
	 * Reason for performing test.
	 */
	public TestReason reasonForTest;
	/**
	 * Identity of lab equipment used to perform test.
	 */
	public String testEquipmentID;
	/**
	 * Specimen on which lab testing done in determining results.
	 */
	public Specimen Specimen;
	/**
	 * Test lab which produced this set of lab test results.
	 */
	public AssetTestLab AssetTestLab;

	public LabTestDataSet(){

	}
}//end LabTestDataSet